README.md

```diff
@@ -1,41 +1,85 @@
+which the container runs.
+
+This repository contains scripts to create a custom
+[root filesystem image](https://github.com/clearcontainers/runtime/blob/master/docs/architecture/architecture.md#root-filesystem-image) ("rootfs") and [guest kernel image](https://github.com/clearcontainers/runtime/blob/master/docs/architecture/architecture.md#guest-kernel). These custom resources may be used for testing and developing new features.
+
+## Introduction
+
+The scripts in this repository are called by running `make(1)` specifying
+particular targets. The scripts use a work directory for storing all files. By
+deafult this will be created automatically as `./workdir` however this can be
+changed by setting the `WORKDIR` environment variable. See [Environment Variables](#Environment Variables).
+
+## Building a guest image
+
+A guest image is a rootfs that has been converted into a disk image.
+
+### Build a rootfs
+
+The `rootfs` target will generate a directory called `workdir/rootfs`,
+overwriting duplicate files:
 
 ```
-sudo -E  make rootfs
+$ sudo -E  make rootfs
 ```
 
-### (optional) customize clear container rootfs ###
-After generate a initial rootfs (previous step) you can 
-customize it by adding anything you need to rootfs 
-directory.
+#### (optional) Customise the rootfs
+
+It is possible to customise the rootfs; simply modify the files below
+`workdir/rootfs` as desired.
+
+#### Create a rootfs image
+
+The `image` target will create a disk image called `container.img` from the `workload/rootfs` directory. This image file is compatible with the official Clear Containers images provided with a Clear Containers installation.
 
-### Create Clear Container image ###
-The `image` target will take the `workload/rootfs` directory
-and will generate a guest image named container.img in the workdir
-directory compatible with Clear Containers. The `image` target will
-not create nor populate the `workdir/rootfs` directory.
-We recommend you use the `rootfs` target before using the `image` target.
-You can easily change the size of the image using the `IMG_SIZE` environment variable.
-See [Environment Variables](#Environment Variables)
+Note:
+
+The `image` target will not create or populate the `workdir/rootfs` directory
+so it is necessary to [build a rootfs](#build-a-rootfs) first.
+
+Use the `IMG_SIZE` environment variable to change the size of the image if
+desired. See [Environment Variables](#Environment Variables).
 
 ```
-sudo -E make image
+$ sudo -E make image
 ```
 
-### Default packages ###
-By default, the root file system for the Clear Containers image is based on
-Clear Linux for Intel\* Architecture, but the `workload/rootfs` directory can be
-populated with any other source. Next, packages are installed inside
-the generated image. You can install extra packages using the
-environment variable `EXTRA_PKGS`. See [Environment Variables](#Environment Variables).
+#### Default packages
```

Comment:
Hi @jcvenegas - does this section need an update for 3.x?

Reply:
@jodh-intel oh yes we need to `cc-agent` to the list.